com.zsmartsystems.zigbee.console/src/main/java/com/zsmartsystems/zigbee/console/ZigBeeConsoleCommandsSupportedCommand.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2016-2018 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package com.zsmartsystems.zigbee.console;
+
+import java.io.PrintStream;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+import com.zsmartsystems.zigbee.ZigBeeEndpoint;
+import com.zsmartsystems.zigbee.ZigBeeNetworkManager;
+import com.zsmartsystems.zigbee.zcl.ZclCluster;
+import com.zsmartsystems.zigbee.zcl.ZclCommand;
+
+/**
+ * Console command that prints the commands that are supported by a given cluster.
+ *
+ * @author Henning Sudbrock
+ */
+public class ZigBeeConsoleCommandsSupportedCommand extends ZigBeeConsoleAbstractCommand {
+
+    @Override
+    public String getCommand() {
+        return "cmdsupported";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Check what commands are supported within a cluster, optionally restricted to generated / received commands";
+    }
+
+    @Override
+    public String getSyntax() {
+        return "ENDPOINT CLUSTER [gen|rcv]";
+    }
+
+    @Override
+    public String getHelp() {
+        return "";
+    }
+
+    @Override
+    public void process(ZigBeeNetworkManager networkManager, String[] args, PrintStream out)
+            throws IllegalArgumentException, InterruptedException, ExecutionException {
+        if (args.length < 3 || args.length > 4) {
+            throw new IllegalArgumentException("Invalid number of arguments");
+        }
+
+        final ZigBeeEndpoint endpoint = getEndpoint(networkManager, args[1]);
+        final Integer clusterId = parseCluster(args[2]);
+        ZclCluster cluster = getCluster(endpoint, clusterId);
+
+        if (cluster == null) {
+            out.println("Cluster not found");
```

Comment:
I thought that in general I was throwing an exception if there was an error? I guess you copied this from the attribute command, so presumably it's not consistent everywhere. What do you think? My original thinking was always to throw exceptions with errors to allow the implementation to catch these given there is no return.

Reply:
I agree that it is nicer to throw an exception than to simply print out a message on the console.